### PR TITLE
Message Fixes

### PR DESCRIPTION
**Pull request by AvarianKnight**
**Pull Request Description**

This provides a fix for read state on messages, as previously they didn't work.

This also includes message ordering based on last activity

This is also a critical fix for group message creations, current `message_group_conversations` will have to be wiped due to a bug  that resulted in them not being created properly

**Pull Request Checklist**:
* [x] Have you followed the guidelines in our Contributing document and Code of Conduct?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/project-error/npwd/pulls) for the same update/change?
* [x] Have you built and tested NPWD in-game after the relevant change?

### CODE REVIEW

File: resources/server/messages/messages.db.ts [hunk 1]
```diff
@@ -147,18 +146,30 @@ export class _MessagesDB {
...
     return result[0].count;
   }
-
+  /**
+   * Adds an unread to the specified group id
+   * @param groupId The unique group ID for the message
+   * @param phoneNumber The phone number of the player to ignore
+   */
+  async setMessageUnread(groupId: string, phoneNumber: string) {
+    console.log(groupId, phoneNumber);
```
Comment (itschip) [inline resources/server/messages/messages.db.ts:155]: Remove

File: resources/server/messages/messages.utils.ts [hunk 2]
```diff
@@ -108,7 +109,7 @@ export async function createMessageGroupsFromPhoneNumber(
       sourcePhoneNumber,
     );
   } else {
-    await MessagesDB.createMessageGroup(sourcePhoneNumber, conversationId, sourcePhoneNumber);
+    await MessagesDB.createMessageGroup(tgtPhoneNumber, conversationId, sourcePhoneNumber);
```
Comment (itschip) [inline resources/server/messages/messages.utils.ts:112]: With this change, have you tested cases like, if you send a message to a person who does not have a conversation with you? Does it create the group properly?

File: resources/server/messages/messages.service.ts [hunk 3]
```diff
@@ -135,7 +143,7 @@ class _MessagesService {
           const participantPlayer = PlayerService.getPlayerFromIdentifier(participantIdentifier);
 
           if (participantPlayer) {
-            emitNet(MessageEvents.SEND_MESSAGE_SUCCESS, participantPlayer.source, messageData);
+            emitNet(MessageEvents.SEND_MESSAGE_SUCCESS, reqObj.source, messageData);
```
Comment (itschip) [inline resources/server/messages/messages.service.ts:146]: What's going on here? We still need to send this to the 'participant'. The broadcast doesn't update any messages.

**Review by itschip — CHANGES_REQUESTED** [review]

File: resources/server/messages/messages.utils.ts (hunk 2 above)
Comment (AvarianKnight) [reply resources/server/messages/messages.utils.ts:112]: Yes it does

File: resources/server/messages/messages.db.ts (hunk 1 above)
Comment (AvarianKnight) [reply resources/server/messages/messages.db.ts:155]: print go brrrrrrrrrrrr

**Pull request closed without merging**